pilot/pkg/config/kube/gateway/deploymentcontroller.go

```diff
@@ -431,6 +432,30 @@ func (d *DeploymentController) apply(controller string, yml string) error {
 	return nil
 }
 
+// canManage checks if a resource we are about to write should be managed by us. If the resource already exists
+// but does not have the ManagedGatewayLabel, we won't overwrite it.
+// This ensures we don't accidentally take over some resource we weren't supposed to, which could cause outages.
+func (d *DeploymentController) canManage(gvr schema.GroupVersionResource, name, namespace string) bool {
+	store, f := d.listers[gvr.GroupVersion().String()]
+	if !f {
+		log.Warnf("unknown GVR %v", gvr)
+		// Even though we don't know what it is, allow users to put the resource. We won't be able to
+		// protect against overwrites though.
+		return true
+	}
+	obj, f, _ := store.GetByKey(controllers.KeyFunc(name, namespace))
+	if f {
+		m, ok := obj.(controllers.Object)
+		if ok {
+			_, managed := m.GetLabels()[constants.ManagedGatewayLabel]
+			// If object already exists, we can only manage it if it has the label
+			return managed
```

Comment:
Do we need to worry about revision/labels ? Or is this protected by a master election - and winner manages everything ?